src/Writer/XLSX/Options.php

```diff
@@ -63,4 +70,22 @@ public function getMergeCells(): array
     {
         return $this->MERGE_CELLS;
     }
+
+    public function setRowAttributes(Row $row, RowAttributes $attributes): Row
+    {
+        $this->rowsAttributes[$row] = $attributes;
+        $this->outlineMaxLevel = max($attributes->getOutlineLevel(), (int) $this->outlineMaxLevel);
+
+        return $row;
+    }
+
+    public function getRowAttributes(Row $row): ?RowAttributes
+    {
+        return $this->rowsAttributes[$row] ?? null;
+    }
```

Comment:
Nullable elements only leads to weak types; let's just return an empty `RowAttributes` and implement an `isEmpty` method on that object
```suggestion
    public function getRowAttributes(Row $row): RowAttributes
    {
        return $this->rowsAttributes[$row] ?? new RowAttributes();
    }
```